clippy.toml

```diff
@@ -14,4 +14,5 @@ allowed-duplicate-crates = [
     "windows_x86_64_gnu",
     "windows_x86_64_gnullvm",
     "windows_x86_64_msvc",
+    "smallvec", # Remove once smallvec 2.0 is released
```

Comment:
Can you add a comment about what the path to duplicates is like above (it's useful to see from a historical perspective why we're facing this and where there might be type incompatibilities.